adal/src/main/java/com/microsoft/aad/adal/AbstractMetadataRequestor.java

```diff
@@ -0,0 +1,81 @@
+package com.microsoft.aad.adal;
+
+import com.google.gson.Gson;
+
+import java.util.UUID;
+
+/**
+ * Creates correlatable Requests to HTTP accessible resources.
+ */
+abstract class AbstractMetadataRequestor<MetadataType, MetadataRequestOptions>
+        implements Correlatable {
```

Comment:
The two requestor objects I've defined (`DRSMetadataRequestor` and `WebFingerMetadataRequestor`) both support the setting/getting of a correlationId. 

While both of these classes have a shared superclass (`AbstractMetadataRequestor`), it felt appropriate to place these two methods in their own interface, as setting/getting correlationIds seems to happen in a few different classes (see [`Discovery`](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/dev/adal/src/main/java/com/microsoft/aad/adal/Discovery.java#L123), [`Logger`](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/dev/adal/src/main/java/com/microsoft/aad/adal/Logger.java#L401))

As this interface is only explicitly implemented by a single class at the moment (`AbstractMetadataRequestor`) it is no trouble for me to declare these getters/setters directly within that class, should that be your preference.

Feel free to make the 'leave-it' or 'move-it' call. I'm satisfied with either.